Firebase/InstanceID/FIRInstanceIDKeyPairStore.m

```diff
@@ -365,31 +365,43 @@ - (void)migrateKeyPairCacheIfNeededWithHandler:(void (^)(NSError *error))handler
   self.keyPair = keyPair;
 
   // Either new key pair doesn't exist or it's different than legacy key pair, start the migration.
+
+  // Use a dispatch group to call handler when both private and public keys are updated.
+  dispatch_group_t dispatchGroup = dispatch_group_create();
```

Comment:
Not sure if this is needed since the local variable keypair is already atomic and the keychain operations are guarded by a serial queue already.

Reply:
Yeah, probably we may call handler from the 2nd keychain handler. It's a bit implicit,  though we already rely on the `FIRInstanceIDKeychain` implementation managed by a serial queue, so, I'll remove dispatch groups.